README.md

```diff
@@ -1,52 +1,205 @@
+An overview of the AMQP concepts and terms is decribed [here](https://www.rabbitmq.com/tutorials/amqp-concepts.html).
+
+
+The following example connects to a queue, registers a Consumer`*` with a handler function and sends two messages to the queue:
+
+```js
+const { Reshuffle } = require('reshuffle')
+const { AMQPConnector } = require('reshuffle-amqp-connector')
+
+  const app = new Reshuffle()
+  const amqp = new AMQPConnector(
+    app, 
+    { queueUrl: process.env.QUEUE_URL // e.g. 'amqp://localhost'
+      queueName: process.env.QUEUE_NAME, 
+      queueOptions: { durable: true } 
+    })
+
+  amqp.on({ noAck: false }, function(msg) {
+      console.log('A new received', msg.content.toString())
+   })
+
+  setTimeout(function(){
+     amqp.sendMessage('MSG-001', { persistent: true })
+     amqp.sendMessage('MSG-002')
+  }, 2000)
+
+```
+
+(`*`) A [Consumer](https://www.rabbitmq.com/consumers.html) is a subscription for the queue. 
+
+(`**`) In the above example the `sendMessage` function is called after a short delay just to enbale the `AMQPConnector` finish all it's internal `async` instansiations.
+
+The `AMQPConnector` is implemented as a [Task Queue](https://www.rabbitmq.com/tutorials/tutorial-two-javascript.html) which means that messages are sent to the queue and one or more Consumers can be registered to this queue. 
+
+You can define several Consumers by calling the `AMQPConnector.on` function multiple times with different `eventId`s.
+
+
+#### Table of Contents
+
+[Configuration](#configuration) Configuration options
+
+[DataTypes](#dataTypes) Data Types
+
+#### Connector Events
+
+[Consuming the queue messages](#consumingMessages) 
+
+[EventsDataTypes](#eventsDataTypes) Events Data Types
+
+#### Connector Actions
+
+[sendMessage](#sendMessage) Send message to queue
+ 
+[sdk](#sdk) Retrieve the client sdk object
+
+
+##### <a name="configuration"></a>Configuration options
 
-#### Configuration Options:
 ```typescript
-interface _CONNECTOR_NAME_ConnectorConfigOptions {
-  foo: string // foo description
-  bar?: number // bar description
+interface AMQPConnectorConfigOptions {
+  queueUrl: string
+  queueName: string
+  queueOptions?: Options.AssertQueue
+}
+```
+
+##### <a name="dataTypes"></a>DataTypes
+
+<a name="_Options.AssertQueue:_"></a>_Options.AssertQueue:_
+
+```ts
+{
+  exclusive?: boolean
+  durable?: boolean
+  autoDelete?: boolean
+  arguments?: any
+  messageTtl?: number
+  expires?: number
+  deadLetterExchange?: string
+  deadLetterRoutingKey?: string
+  maxLength?: number
+  maxPriority?: number
 }
 ```
+The `Options` namespace can be imported form the [`amqplib`](https://www.npmjs.com/package/amqplib) client.
+
+More details about Options.AssertQueue available [here](http://www.squaremobius.net/amqp.node/channel_api.html#channel_assertQueue).
+
 
 #### Connector events
 
-##### event1 description
-The connector fires this event when ...
+##### <a name="consumingMessages"></a>Consuming the queue messages
```

Comment:
The event is singular. So it is really a queue message, not consuming the queue messages.